pkg/certificate/manager.go

```diff
@@ -190,6 +191,42 @@ func (m *Manager) rotateServices() error {
 	return nil
 }
 
+// nextRotationDeadlineForService will look at the first service at
+// webhook configuration find the secret's TLS certificate and calculate
+// next deadline, looking at first serices is fine since they certificates
+// are created/rotated at the same time
+func (m *Manager) nextRotationDeadlineForServices() time.Time {
+	webhookConf, err := m.readyWebhookConfiguration()
+	if err != nil {
+		m.log.Info(fmt.Sprintf("failed getting webhook configuration, forcing rotation: %v", err))
+		return m.now()
+	}
+
+	services, err := m.getServicesFromConfiguration(webhookConf)
+	if err != nil {
+		m.log.Info(fmt.Sprintf("failed getting webhook configuration services, forcing rotation: %v", err))
+		return m.now()
+	}
+
+	// Iterate the `services` map to calculate deadline with the first
+	// occurrence
+	for service, _ := range services {
```

Comment:
If I understand this correctly, then right now we run a for loop of the services, but we return the first one.
Shouldn't we return the one with the earliest deadline out of the services?

Reply:
We create service certificates at the same time so they have almost the same deadline so no need for this.

Reply:
yes but it's a private case behavior that might change in the future. I think we should keep it general, i.e., go over all services and take the earliest one.

Reply:
fair enough.